### PR TITLE
appointments validation

### PR DESCRIPTION
**Pull request by nintendaii**
Modules used:
-@hapi/joi
-@hapi/joi-date
-moment

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -6,6 +6,13 @@ var config = require('./config');
 
 var mongoose = require('mongoose');
 
+app.use(function(req, res, next) {
```
Comment (user4i) [inline main.js]: Please use cors package for 9-14 lines

File: package.json [hunk 2]
```diff
@@ -16,6 +16,8 @@
   },
   "homepage": "https://github.com/appointus/api#readme",
   "dependencies": {
+    "@hapi/joi": "^15.1.0",
```
Comment (user4i) [inline package.json]: use moment for it not @hapi/joi, @hapi/joi-date

**Review by user4i — CHANGES_REQUESTED** [review]

File: routes/appointments.js [hunk 3]
```diff
@@ -9,6 +14,12 @@ router.get('/appointments/:date', function(req, res) {
 });
 
 router.post('/appointments', function(req, res) {
+  var currentDate = moment().format('YYYY-MM-DD');
+  var currentTime = moment().format('H:mm');
+  const { error } = validateAppointment(req.body); //object destructuring , the same as result.error
+  if (error) return res.status(400).send(error.details[0].message);
+  if (currentDate > req.body.date || (currentDate == req.body.date && currentTime > req.body.time))
```
Comment (user4i) [inline routes/appointments.js]: And what about moment.isAfter, moment.isBefore?

**Review by user4i — CHANGES_REQUESTED** [review]

File: main.js (hunk 1 above)
Comment (nintendaii) [reply main.js]: done

**Pull request merged**